json.c

```diff
@@ -42,17 +42,11 @@ const struct _json_value json_value_none;
 #include <string.h>
 #include <ctype.h>
 #include <math.h>
+#include <limits.h>
 
 typedef unsigned int json_uchar;
 
-/* There has to be a better way to do this */
-static const json_int_t JSON_INT_MAX = sizeof(json_int_t) == 1
-                                       ? INT8_MAX
-                                       : (sizeof(json_int_t) == 2
-                                         ? INT16_MAX
-                                         : (sizeof(json_int_t) == 4
-                                           ? INT32_MAX
-                                           : INT64_MAX));
+#define JSON_INT_MAX ((1UL << ((sizeof(json_int_t) * CHAR_BIT) - 1)) - 1)
```

Comment:
For a patch release such as 1.1.1, I would change as little code as possible:
```C
static const json_int_t JSON_INT_MAX = (json_int_t)((1UL << ((sizeof(json_int_t) * CHAR_BIT) - 1)) - 1);
```

Reply:
Also, `1UL` is an `unsigned long`. What if `json_int_t` is larger than `unsigned long`?

Reply:
That's a very good point, perhaps the alternative you proposed in #84 would be better then